src/libstore/binary-cache-store.cc

```diff
@@ -57,11 +57,12 @@ void BinaryCacheStore::init()
             auto value = trim(line.substr(colon + 1, std::string::npos));
             if (name == "StoreDir") {
                 if (value != storeDir)
-                    throw Error(
+                    warn(
                         "binary cache '%s' is for Nix stores with prefix '%s', not '%s'",
                         config.getHumanReadableURI(),
                         value,
                         storeDir);
+                config.storeDir = value;
```

Comment:
I would love to have some systematic stuff for local vs remote setting, fyi. E.g. if you explicitly do `?store=...`, you probably don't want the remote side to silently overwrite that.